common/remote/hooks/libarchive/archive.cpp

```diff
@@ -608,7 +608,7 @@ class CArchiveFileHook : public CInterface, implements IContainedFileHook
 
 extern ARCHIVEFILE_API void installFileHook()
 {
-    SpinBlock b(*lock); // Probably overkill!
+    CriticalBlock b(*lock); // Probably overkill!
```

Comment:
My current position (subject to change!) is that we should only be using critical sections when the code is very time critical, and the work done in the critical section is as small as possible.

The spin locks have two problems:  On multiple socket systems under high contention they perform badly.  More significantly if the number of threads exceeds the number of cores they can perform terribly if the thread that holds the lock is descheduled.

So they have a relatively minor benefit if we can guarantee low contention and little work inside them, but a potentially bad worse-case behaviour.  Unless there is a clear benefit I think we should err on the side of critical sections.